### PR TITLE
Allow creating permalinks for unknown predecessor rooms (even though unlikely to be successful)

### PR DESCRIPTION
**Pull request by justjanne**
Type: Defect
Partially fixes: https://github.com/vector-im/element-web/issues/24453
Reopens: https://github.com/vector-im/element-web/issues/2925

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Allow creating permalinks for unknown predecessor rooms (even though unlikely to be successful) ([\#10113](https://github.com/matrix-org/matrix-react-sdk/pull/10113)). Fixes vector-im/element-web#24453. Contributed by @justjanne.

### CODE REVIEW

**Comment by t3chguy** [comment]
If this lands I think it warrants reopening https://github.com/vector-im/element-web/issues/2925 as that was closed as everything moved to vias, this regresses that

File: src/components/views/messages/RoomPredecessorTile.tsx [hunk 1]
```diff
@@ -86,12 +86,12 @@ export const RoomPredecessorTile: React.FC<IProps> = ({ mxEvent, timestamp }) =>
...
-    if (!prevRoom) {
-        logger.warn(`Failed to find predecessor room with id ${predecessor.roomId}`);
-        return <></>;
-    }
     const permalinkCreator = new RoomPermalinkCreator(prevRoom, predecessor.roomId);
-    permalinkCreator.load();
+    if (prevRoom) {
+        permalinkCreator.load();
+    } else {
+        logger.warn(`Creating permalink for unknown predecessor room ${predecessor.roomId}, unlikely to be successful`);
```
Comment (MadLittleMods) [inline src/components/views/messages/RoomPredecessorTile.tsx:93]:
```suggestion
        logger.warn(`Creating permalink for unknown predecessor room ${predecessor.roomId}, it's possible that this room doesn't actually exist but we will give the user what they want`);
```

Comment (MadLittleMods) [reply src/components/views/messages/RoomPredecessorTile.tsx:93]:
```suggestion
        logger.warn(`Creating permalink for unknown predecessor room ${predecessor.roomId}, unlikely to be routable on its own without via servers but we will give the user what they want`);
```

Comment (justjanne) [reply src/components/views/messages/RoomPredecessorTile.tsx:93]:
```suggestion
        logger.warn(`Creating permalink for unknown predecessor room ${predecessor.roomId},`+
            'unlikely to be successful as rooms without via servers are usually not routable');
```

**Pull request closed without merging**